pkgs/games/srb2kart/default.nix

```diff
@@ -0,0 +1,128 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, fetchzip
+, cmake
+, pkg-config
+, nasm
+, makeDesktopItem
+, copyDesktopItems
+, SDL2
+, SDL2_mixer
+, zlib
+, libopenmpt
+, libgme
+, libpng
+, curl
+}:
+
+let
+  version = "1.3";
+
+  assets = fetchzip {
+    name = "srb2kart-assets";
+    url = "https://github.com/STJr/Kart-Public/releases/download/v${version}/srb2kart-v${lib.replaceStrings ["."] [""] version}-Installer.exe";
+    sha256 = "sha256-WDE7noJTy+JLsSh5ETcTFe1b4jz2RgZ/zuYFkQ2U+w8=";
+    extension = "zip";
+    stripRoot = false;
+    extraPostFetch = ''
+      rm $out/*.dll $out/*.exe $out/*.bat
+    '';
+  };
+
+  description = "A kart racing mod based on the 3D Sonic the Hedgehog fangame Sonic Robo Blast 2";
+  desktopItem = makeDesktopItem {
+    name = "srb2kart";
+    desktopName = "Sonic Robo Blast 2 Kart";
+    comment = description;
+    exec = "srb2kart";
+    icon = "srb2kart";
+    terminal = false;
+    type = "Application";
+    startupNotify = false;
+    categories = [ "Application" "Game" ];
+  };
+
+in stdenv.mkDerivation rec {
+  pname = "srb2kart";
+  inherit version;
+
+  # There is also a GitHub mirror, but their own gitlab seems to be the main one
+  # ...but assets are only published on GitHub
+  src = fetchFromGitLab {
+    domain = "git.do.srb2.org";
+    owner = "KartKrew";
+    repo = "Kart-Public";
+    rev = "v${version}";
+    sha256 = "sha256-iXh0RPnt6Zzt8aj+HX4dHU57Qft+ME0n+BvGxOpKL4w=";
+  };
+
+  patches = [
+    # We add a reference to the assets derivation in postPatch
+    # TODO send upstream a patch to add SRB2_ASSET_INSTALL here too
+    ./dont-install-assets.patch
+  ];
+
+  postPatch = ''
+    # '#include "SDL_thing.h"' -> '#include <SDL2/SDL_thing.h>'
+    grep -Zlr '#include \+"SDL' src | xargs -0 sed -ie \
+      's@#include \+"SDL\(_[a-zA-Z]\+\)\?.h"@#include <SDL2/SDL\1.h>@'
+    sed -ie 's@<SDL.h>@<SDL2/SDL.h>@' src/sdl/i_threads.c
```

Comment:
If this would break I couldn't fix it. Is there an easier way?

Reply:
If I use -E I can remove backslashes from the sed expression: `s@#include +"SDL(_[a-zA-Z]+)?.h"@#include <SDL2/SDL\1.h>@`

Or I could make it less strict: `s@#include +"SDL(.*).h"@#include <SDL2/SDL\1.h>@`

Would one of those be fine?

Reply:
Maybe a patch would be easier? In general I think it is always a good idea to avoid very complex shell structures if the same can be easily achieved with another methord.